lib/spack/spack/bootstrap.py

```diff
@@ -346,6 +506,8 @@ def get_executable(exe, spec=None, install=False):
     install quickly (when using external python) or are guaranteed by Spack
     organization to be in a binary mirror (clingo).
     """
+    import spack.user_environment as uenv
```

Comment:
Was this really circular without this being moved here? I'm having trouble figuring out where the circle would be.